sources/packages/api/src/services/designation-agreement/designation-agreement.service.ts

```diff
@@ -147,4 +163,114 @@ export class DesignationAgreementService extends RecordDataModelService<Designat
       .getRawOne();
     return !!found;
   }
+
+  /**
+   * Service to validate designationId passed onto API.
+   * @param designationId
+   * @returns flag that indicates if the designationId is valid data.
+   */
+  async designationForUpdateExist(designationId: number): Promise<boolean> {
+    const found = await this.repo
+      .createQueryBuilder("designation")
+      .select("1")
+      .where("designation.id = :designationId", { designationId })
+      .andWhere("designation.designationStatus != :designationStatus", {
+        designationStatus: DesignationAgreementStatus.Declined,
+      })
+      .getRawOne();
+    return !!found;
+  }
+
+  /**
+   * Update designation for Approval/Denial or re-approve.
+   * @param designationId Designation which is going to be updated.
+   * @param userId User who updates the designation.
+   * @param designationPayload update payload.
+   */
+  async updateDesignation(
+    designationId: number,
+    userId: number,
+    designationPayload: UpdateDesignationDto,
+  ): Promise<void> {
+    const designation = new DesignationAgreement();
+    designation.id = designationId;
+    designation.designationStatus = designationPayload.designationStatus;
+    designation.startDate = designationPayload.startDate;
+    designation.endDate = designationPayload.endDate;
+    designation.assessedBy = { id: userId } as User;
+    designation.assessedDate = new Date();
+    designation.designationAgreementLocations =
+      designationPayload.locationsDesignations?.map(
+        (locationPayload: UpdateDesignationLocationDto) => {
+          const location = new DesignationAgreementLocation();
+          location.id = locationPayload.designationLocationId;
+          location.requested = locationPayload.requested;
```

Comment:
Is there a scenario where this value will be updated other then during the record creation?
I am under the impression that this should not be part of the update.

Reply:
This value is moved to be only under record creation part if ministry is approving location that is not requested.

Reply:
location.requested removed.